test/metabase/pulse/pulse_integration_test.clj

```diff
@@ -166,6 +166,17 @@
                                                 :user_id          (mt/user->id :rasta)}]
           (is (all-pct-2d? (first (run-pulse-and-return-last-data-columns pulse)))))))))
 
+(defn- strip-timestamp
+  "Remove the timestamp portion of attachment filenames.
+  This is useful for creating stable filename keys in tests.
+  For example, see `run-pulse-and-return-attached-csv-data` below.
+
+  Eg. test_card_2024-03-05T22:30:24.077306Z.csv -> test_card.csv"
+  [fname]
+  (let [ext (last (str/split fname #"\."))
+        name-parts (butlast (str/split fname #"_"))]
+    (format "%s.%s" (str/join "_" name-parts) ext)))
```

Comment:
Will a date ever have an _ in it? Assuming no. Just want to make sure this is a good splitter. This is probably good, though.

Reply:
Hmm... yea good point. I think it's fine though. I guess if some tests fail in the future this can be adjusted as needed.